x-pack/plugins/enterprise_search/public/applications/app_search/index.test.tsx

```diff
@@ -69,30 +69,83 @@ describe('AppSearchConfigured', () => {
   it('does not re-initialize app data', () => {
     const initializeAppData = jest.fn();
     (useActions as jest.Mock).mockImplementation(() => ({ initializeAppData }));
-    (useValues as jest.Mock).mockImplementation(() => ({ hasInitialized: true }));
+    (useValues as jest.Mock).mockImplementation(() => ({ myRole: {}, hasInitialized: true }));
 
     shallow(<AppSearchConfigured />);
 
     expect(initializeAppData).not.toHaveBeenCalled();
   });
 
   it('renders ErrorConnecting', () => {
-    (useValues as jest.Mock).mockImplementation(() => ({ errorConnecting: true }));
+    (useValues as jest.Mock).mockImplementation(() => ({ myRole: {}, errorConnecting: true }));
 
     const wrapper = shallow(<AppSearchConfigured />);
 
     expect(wrapper.find(ErrorConnecting)).toHaveLength(1);
   });
+
+  describe('ability checks', () => {
+    it('renders a 404 if a user has no view abilities', () => {
+      const wrapper = shallow(<AppSearchConfigured />);
+
+      expect(wrapper.find(NotFound)).toHaveLength(2);
+    });
+
+    it('renders the engines overview if a user can view engines', () => {
+      (useValues as jest.Mock).mockImplementation(() => ({
+        myRole: { canViewEngines: true },
+      }));
+
+      const wrapper = shallow(<AppSearchConfigured />);
+
+      expect(wrapper.find(EngineOverview)).toHaveLength(1);
```

Comment:
How are the EngineOverview AND the NotFound both present? 🤔

Reply:
NotFound is the generic fallback for all routes:

https://github.com/elastic/kibana/blob/e17d61d6133e92a5a6673fc3051066ccef5f91b0/x-pack/plugins/enterprise_search/public/applications/app_search/index.tsx#L86-L88

So it's always present / never wrapped in a conditional.